src/course-search/SearchContext.jsx

```diff
@@ -53,11 +53,14 @@ const SearchData = ({ children, searchFacetFilters }) => {
     }
   }, [newQueryString]);
 
-  const value = {
-    refinementsFromQueryParams,
-    dispatch,
-    searchFacetFilters,
-  };
+  const value = useMemo(
```

Comment:
Note: Context values should always be memoized to avoid unnecessary re-rendering of children components that rely on the context values.